molecularnodes/io/universe/ui.py

```diff
@@ -0,0 +1,112 @@
+"""
+Importing molecular dynamics trajectories and associated files.
+"""
+
+__name__ = "MolecularNodes.trajectory"
+__author__ = "Brady Johnston"
+
+import bpy
+import MDAnalysis as mda
+
+from ...blender import path_resolve
+from .universe import MNUniverse
+from bpy.props import StringProperty
+
+bpy.types.Scene.MN_import_md_topology = StringProperty(
+    name="Topology",
+    description="File path for the toplogy file for the trajectory",
+    subtype="FILE_PATH",
+    maxlen=0,
+)
+bpy.types.Scene.MN_import_md_trajectory = StringProperty(
+    name="Trajectory",
+    description="File path for the trajectory file for the trajectory",
+    subtype="FILE_PATH",
+    maxlen=0,
+)
+bpy.types.Scene.MN_import_md_name = StringProperty(
+    name="Name",
+    description="Name of the molecule on import",
+    default="NewTrajectory",
+    maxlen=0,
+)
+
+
+def load(
+    top,
+    traj,
+    name="NewTrajectory",
+    style="spheres",
+    subframes: int = 0,
+):
+    top = path_resolve(top)
+    traj = path_resolve(traj)
+
+    universe = mda.Universe(top, traj)
+
+    mn_universe = MNUniverse(universe=universe)
+
+    mn_universe.create_model(name=name, style=style, subframes=subframes)
+
+    return mn_universe
+
+
+class MN_OT_Import_Protein_MD(bpy.types.Operator):
```

Comment:
Just an idea for a more OOP design: I think there are a couple of similar code segments for importing in the UI. Maybe it's a better design to have a base class `MN_OT_Import_Base` for all of them, with a function `load` for each class. Also, I think `load` is doing dual jobs here of both loading and creating the model so maybe separating them into two functions as well.

Reply:
When I first created the add-on I wasn't holding on to the python objects after import, I was having a single function for each import method that would load and create the object. Now that everything has it's own class that hangs around, I think removing the `load` functions for everything makes more sense. 

Once you parse something into one of the classes, the user can call the `create_model()` or `create_object()` function (unsure about naming) to then create the actual object associated with the class / model / universe / molecule / etc. I want to keep the operator calls simple, so having the contents for each of the `load` functions just be the operator `execture()` functions I think makes sense and simplifies the code base a bit. 

I do however like the idea of this as the lines required to import and create a model into a scene.
```py
import molecularnodes as mn
mn.io.fetch('1cd3')
```

This is slightly more verbose, but I suppose more 'pythonic' and a better representation of the data flow in the scene.
```py
import molecularnodes as mn
mol = mn.io.fetch('1cd3')
mol.create_model()
```